src/control/common/collection_utils.go

```diff
@@ -9,12 +9,49 @@ package common
 import (
 	"encoding/json"
 	"fmt"
+	"sort"
 	"strings"
 	"unicode"
 
 	"github.com/pkg/errors"
 )
 
+// StringSet is a non-duplicated set of strings.
+type StringSet map[string]interface{}
```

Comment:
I think it's fine to add this to the common package as it's generally useful. I have a minor quibble with the implementation, though. As the only purpose of the map is to indicate existence of the string in the map/set, the value here should be an empty struct. It is slightly more efficient in that there are zero allocations done for each value, but more importantly, there is zero ambiguity about how this type can be used.

I also recommend adding a method to check for membership of a string, so that users of this type do not need to concern themselves with the implementation details. e.g.
```go
func (s StringSet) Has(str string) bool {
        _, exists := s[str]
        return exists
}

Reply:
I think this was added by @kjacque

Reply:
Yeah, this is from my patch. I can make the change in my follow-on PR.